pkg/registry/apis/query/register.go

```diff
@@ -87,6 +92,49 @@ func NewQueryAPIBuilder(features featuremgmt.FeatureToggles,
 	}, nil
 }
 
+func NewLateHeadersSuppliedQueryAPIBuilder(features featuremgmt.FeatureToggles,
```

Comment:
Leaves the original one in place because it's in use by the MT querier in enterprise and has its own way of propagating headers - something I thought would render this newer signature mostly useless for its uses.